cartoframes/data/clients/auth_api_client.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import absolute_import
+
+from carto.api_keys import APIKeyManager
+
+from ...auth import get_default_credentials
+
+
+class AuthAPIClient(object):
+    """AuthAPIClient class is a client of the CARTO Auth API.
+    More info: https://carto.com/developers/auth-api/
+
+    Args:
+        credentials (:py:class:`Credentials <cartoframes.auth.Credentials>`, optional):
+              credentials of user account to send Dataset to. If not provided,
+              a default credentials (if set with :py:meth:`set_default_credentials
+              <cartoframes.auth.set_default_credentials>`) will attempted to be
+              used.
+    """
+    def __init__(self, credentials=None):
+        credentials = credentials or get_default_credentials()
+        self._api_key_manager = _get_api_key_manager(credentials)
```

Comment:
Filtering the API keys with the one provided by the user to check if it is the master one? This also applies to the Kuviz creation, the user should use the master one.

I am not sure about it. Maybe we can catch `Access denied` exceptions and remembering the user about using the master key?

Reply:
Anyway, the validation of the credentials is something to work on because we are not validating is the credentials are defined or not (we are doing the classical `credentials or get_default_credentials()`, but not checking if we have credentials or not). Maybe we can create a task with both objectives

Reply:
https://github.com/CartoDB/cartoframes/issues/1097